rskj-core/src/main/java/org/ethereum/db/BlockStoreEncoder.java

```diff
@@ -0,0 +1,95 @@
+package org.ethereum.db;
+
+import java.util.Optional;
+
+import org.ethereum.core.Block;
+import org.ethereum.core.BlockFactory;
+import org.ethereum.core.BlockHeader;
+import org.ethereum.util.RLP;
+import org.ethereum.util.RLPList;
+
+/**
+ * Encodes and decodes both block headers and blocks allowing both value types to be stored the same block store.
+ */
+public class BlockStoreEncoder {
+
+    private final BlockFactory blockFactory;
+
+    /**
+     * Creates a new block store encoder.
+     *
+     * @param blockFactory The block factory is needed to decode encoded values, cannot be null.
+     */
+    public BlockStoreEncoder(BlockFactory blockFactory) {
+        this.blockFactory = blockFactory;
+    }
+
+    /**
+     * Encodes a header to store in the block store.
+     *
+     * @param blockHeader The block header to encode, cannot be null.
+     * @return The encoded header, never null.
+     */
+    public byte[] encodeBlockHeader(BlockHeader blockHeader) {
+        if (blockHeader == null) {
```

Comment:
Use `Objects.requireNonNull`

Reply:
The thing about requireNonNull is that it throws a null pointer exception, I think of a null value in this situation as an IllegalArgument